system/lib/wasmfs/backends/map_backend.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2025 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+// This file defines the fetch backend.
+
+#include "backend.h"
+#include "wasmfs.h"
+#include "virtual.h"
+#include "file.h"
+#include "paths.h"
+#include "memory_backend.h"
+#include <sstream>
+
+namespace wasmfs {
+
+typedef std::map<std::string,std::string> MapManifest;
+
+class MapBackend : public wasmfs::Backend {
+  MapManifest *manifest;
+ public:
+  // Takes ownership of manifest
+  MapBackend(MapManifest *manifest): manifest(manifest) {
+    assert(manifest && "Mapfs: null manifest not supported");
+  }
+  ~MapBackend() {
+    if(manifest != NULL) {
+      delete manifest;
+    }
+  }
+  std::shared_ptr<DataFile> createFile(mode_t mode) override;
+  std::shared_ptr<Directory> createDirectory(mode_t mode) override;
+  std::shared_ptr<Symlink> createSymlink(std::string target) override {
+    fprintf(stderr, "mapfs doesn't support creating symlinks");
```

Comment:
```suggestion
    fprintf(stderr, "MAPFS doesn't support creating symlinks");
```
I think this is more consistent with other text?